### PR TITLE
V8: Use the content type model when creating a "matching template"

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/5152

### Description

As @nielslyngsoe points out in #5152, the "Create matching template" functionality creates generic templates (same as when you create a template manually) - not content type specific ones.  And that's silly, since we know what content type the template is created for 😄 

This PR fixes it:

matching-template-with-model

#### Testing this PR

1. Create a content type without a template and save it (it doesn't need any properties).
2. Hit "Create matching template" for the content type.
3. Open the created template and verify that it's generated for the specific content type - i.e. it inherits `@inherits Umbraco.Web.Mvc.UmbracoViewPage<[content type model class]>`

### CODE REVIEW

File: src/Umbraco.Web/Editors/ContentTypeController.cs [hunk 1]
```diff
@@ -322,6 +322,23 @@ public DocumentTypeDisplay PostSave(DocumentTypeSave contentTypeSave)
             return display;
         }
 
+        public TemplateDisplay PostCreateDefaultTemplate(int id)
+        {
+            var contentType = Services.ContentTypeService.Get(id);
+            if (contentType == null)
+            {
+                throw new NullReferenceException("No content type found with id " + id);
```
Comment (Shazwazza) [inline src/Umbraco.Web/Editors/ContentTypeController.cs]: Should return a 404

File: src/Umbraco.Web/Editors/ContentTypeController.cs [hunk 2]
```diff
@@ -322,6 +322,23 @@ public DocumentTypeDisplay PostSave(DocumentTypeSave contentTypeSave)
...
+            }
+
+            var template = CreateTemplateForContentType(contentType.Alias, contentType.Name);
+            if (template == null)
+            {
+                throw new NullReferenceException("Could not create default template for content type with id " + id);
```
Comment (Shazwazza) [inline src/Umbraco.Web/Editors/ContentTypeController.cs]: We don't really want to ever throw a NullReferenceException, that exception should be reserved by .NET itself and only used to indicate that you have a null object reference.  In this case it should be InvalidOperationException, see https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions

File: src/Umbraco.Web/Editors/ContentTypeController.cs (hunk 1 above)
Comment (kjac) [reply src/Umbraco.Web/Editors/ContentTypeController.cs]: 👍

File: src/Umbraco.Web/Editors/ContentTypeController.cs (hunk 2 above)
Comment (kjac) [reply src/Umbraco.Web/Editors/ContentTypeController.cs]: Gotcha. I was just coding for consistency, given that NullReferenceException is used elsewhere in `ContentController` - but when I look closer it also uses `HttpResponseException` so my bad 👍 

Both exceptions are fixed in 1aec0e1

**Comment by elit0451** [comment]
Requested changes have been made 👍 Tested and verified that it works -> merging @kjac

**Pull request merged**